# origin: Debian (#271533)
# cset: n/a
# inclusion: sent to upstream
# description: add missing include to asm-alpha/unistd.h
# revision date: 2004-09-16

I'm trying to build alsa-modules from alsa source for
kernel-image-2.4.27-1-generic (sarge). If I try to build support for all
cards, it stops with an error in <asm-alpha/unistd.h>.

It turns out that in <asm-alpha/unistd.h> a struct pt_regs regs is
declared without including <asm-alpha/ptrace.h> where it is defined. In 2.6.8
this is fixed (it includes ptrace.h plus a bunch of others).

So, I propose this fix:

--- a/include/asm-alpha/unistd.h	2004-09-13 12:13:22.000000000 +0200
+++ b/include/asm-alpha/unistd.h	2004-09-13 12:14:49.000000000 +0200
@@ -519,6 +519,7 @@

 #include <linux/string.h>
 #include <linux/signal.h>
+#include <asm/ptrace.h>

 extern void sys_idle(void);
 static inline void idle(void)


-- System Information:
Debian Release: 3.1
  APT prefers testing
  APT policy: (500, 'testing')
Architecture: alpha
Kernel: Linux 2.4.27-1-generic
Locale: LANG=C, LC_CTYPE=C

Versions of packages kernel-image-2.4.27-1-generic depends on:
ii  coreutils [fileutils]         5.2.1-2    The GNU core utilities
ii  initrd-tools                  0.1.74     tools to create initrd image for p
ii  modutils                      2.4.26-1   Linux module utilities

-- no debconf information


-- 
To UNSUBSCRIBE, email to debian-kernel-REQUEST@lists.debian.org
with a subject of "unsubscribe". Trouble? Contact listmaster@lists.debian.org

